### PR TITLE
`azurerm_spring_cloud_api_portal` - split to create/update functions

### PR DESCRIPTION
**Pull request by ms-henglu**
```
=== RUN   TestAccSpringCloudAPIPortal_basic
=== PAUSE TestAccSpringCloudAPIPortal_basic
=== CONT  TestAccSpringCloudAPIPortal_basic
=== RUN   TestAccSpringCloudAPIPortal_requiresImport
=== PAUSE TestAccSpringCloudAPIPortal_requiresImport
=== CONT  TestAccSpringCloudAPIPortal_requiresImport
=== RUN   TestAccSpringCloudAPIPortal_complete
=== PAUSE TestAccSpringCloudAPIPortal_complete
=== CONT  TestAccSpringCloudAPIPortal_complete
=== RUN   TestAccSpringCloudAPIPortal_update
=== PAUSE TestAccSpringCloudAPIPortal_update
=== CONT  TestAccSpringCloudAPIPortal_update
--- PASS: TestAccSpringCloudAPIPortal_requiresImport (846.38s)
--- PASS: TestAccSpringCloudAPIPortal_update (849.44s)
--- PASS: TestAccSpringCloudAPIPortal_complete (904.65s)
--- PASS: TestAccSpringCloudAPIPortal_basic (1033.62s)
PASS

```

### CODE REVIEW

File: internal/services/springcloud/spring_cloud_api_portal_resource.go [hunk 1]
```diff
@@ -185,6 +185,75 @@ func resourceSpringCloudAPIPortalCreateUpdate(d *pluginsdk.ResourceData, meta in
...
+	springId, err := parse.SpringCloudServiceID(d.Get("spring_cloud_service_id").(string))
+	if err != nil {
+		return err
+	}
+	id := parse.NewSpringCloudAPIPortalID(subscriptionId, springId.ResourceGroup, springId.SpringName, d.Get("name").(string))
+
+	existing, err := client.Get(ctx, id.ResourceGroup, id.SpringName, id.ApiPortalName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("retreiving %s: %+v", id, err)
```
Comment (stephybun) [inline internal/services/springcloud/spring_cloud_api_portal_resource.go:203]: This is misspelled multiple times, please fix all of them
```suggestion
			return fmt.Errorf("retrieving %s: %+v", id, err)
```

File: internal/services/springcloud/spring_cloud_api_portal_resource.go [hunk 2]
```diff
@@ -185,6 +185,75 @@ func resourceSpringCloudAPIPortalCreateUpdate(d *pluginsdk.ResourceData, meta in
...
+		Properties: properties,
+		Sku:        sku,
+	}
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.SpringName, id.ApiPortalName, apiPortalResource)
+	if err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation/update of %s: %+v", id, err)
```
Comment (stephybun) [inline internal/services/springcloud/spring_cloud_api_portal_resource.go:250]: The error messaged in the create function also need to be updated.
```suggestion
		return fmt.Errorf("updating %s: %+v", id, err)
	}

	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
		return fmt.Errorf("waiting for update of %s: %+v", id, err)
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @ms-henglu. Tests look fine, but there are some typos in the error messages.

**Comment by ms-henglu** [comment]
Hi @stephybun ,

Thanks for the code review! I've updated this PR as suggested, please take another look.

**Review by stephybun — APPROVED** [review]
Thanks @ms-henglu LGTM 👍

**Pull request merged**